File: python/532.py
class Solution:
    def findPairs(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        if k < 0:
            return 0
        ans = 0
        d = {}
        nums.sort()
        for i in nums:
            d[i] = d.get(i,0) + 1

        for i in d :
            if k != 0:
                if (i+k) in d:
                    ans += 1
            else:
                if d[i] > 1:
                    ans += 1
        return ans

File: java/532.java
class Solution {
    public int findPairs(int[] nums, int k) {
        Arrays.sort(nums);
        int slow = 0;
        int fast = 1;
        int ans = 0;
        int prev = Integer.MAX_VALUE;
        while(fast < nums.length){
            while(slow < fast && nums[fast]-nums[slow]>k){
                slow ++;
            }
            if(slow != fast && prev != nums[slow] && nums[fast] - nums[slow] == k){
                ans ++;
                prev = nums[slow];
            }
            fast ++;
        }
        return ans;
    }
}